### PR TITLE
Changed hentaifoundry file name format

### PR DESCRIPTION
**Pull request by cyian-1756**
test link 

http://www.hentai-foundry.com/pictures/user/Moonbound

### CODE REVIEW

File: src/main/java/com/rarchives/ripme/ripper/rippers/HentaifoundryRipper.java [hunk 1]
```diff
@@ -94,16 +94,18 @@ public Document getNextPage(Document doc) throws IOException {
                 logger.info("Couldn't find user & image ID in " + thumb.attr("href"));
                 continue;
             }
+            String[] titleSplit = thumb.attr("herf").split("/");
```
Comment (metaprime) [inline src/main/java/com/rarchives/ripme/ripper/rippers/HentaifoundryRipper.java]: > `herf`
typo in the patch or on the site or what?

File: src/main/java/com/rarchives/ripme/ripper/rippers/HentaifoundryRipper.java [hunk 2]
```diff
@@ -94,16 +94,18 @@ public Document getNextPage(Document doc) throws IOException {
...
+            String title = titleSplit[titleSplit.length -1];
             String user = imgMatcher.group(1),
                 imageId = imgMatcher.group(2);
             String image = "http://pictures.hentai-foundry.com//";
             image += user.toLowerCase().charAt(0);
-            image += "/" + user + "/" + imageId + ".jpg";
+            image += "/" + user + "/" + imageId + "/" + user + "-" + imageId + title + ".png";
```
Comment (metaprime) [inline src/main/java/com/rarchives/ripme/ripper/rippers/HentaifoundryRipper.java]: > `.png`

Strictly PNGs now? String JPG before? Or is this like imgur where file ext on the url doesn't matter and it serves up something reasonable?

**Comment by metaprime** [comment]
The example album works for me, but just want clarification on the above comments before merging.

File: src/main/java/com/rarchives/ripme/ripper/rippers/HentaifoundryRipper.java (hunk 2 above)
Comment (cyian-1756) [inline src/main/java/com/rarchives/ripme/ripper/rippers/HentaifoundryRipper.java]: As far as I can tell it's strictly png now and was strictly JPG before (I ripped a few albums at random and they all were .png) 

> Or is this like imgur where file ext on the url doesn't matter and it serves up something reasonable?

Nope. It 404s if you get the file type wrong

File: src/main/java/com/rarchives/ripme/ripper/rippers/HentaifoundryRipper.java (hunk 1 above)
Comment (cyian-1756) [inline src/main/java/com/rarchives/ripme/ripper/rippers/HentaifoundryRipper.java]: Typo in the patch

File: src/main/java/com/rarchives/ripme/ripper/rippers/HentaifoundryRipper.java (hunk 2 above)
Comment (metaprime) [inline src/main/java/com/rarchives/ripme/ripper/rippers/HentaifoundryRipper.java]: So it's all PNG now?

Comment (cyian-1756) [inline src/main/java/com/rarchives/ripme/ripper/rippers/HentaifoundryRipper.java]: > So it's all PNG now?

Yes

**Comment by cyian-1756** [comment]
@metaprime I added ripping for all file types in my latest commit

**Comment by metaprime** [comment]
Sorry for the delay on this one. It looks like the site is now behind an age wall. Doesn't rip for me now, when going to the site I have to click through an age wall. Might need a cookie to get through. Could you check into it to see if there's another way?

**Comment by metaprime** [comment]
Okay sorry, I'm dumb. It's always been behind an age wall. The code accounted for that. However, it seems the way this ripper handles it isn't working anymore.

I tried a few things but I wasn't able to come up with a quick fix.

**Comment by cyian-1756** [comment]
@metaprime I'll checkout how the age-wall has changed and see if I can come up with a fix

**Comment by cyian-1756** [comment]
@metaprime the age-wall hasn't changed at all (It still uses your phpsession cookie) but the site layout has changed, I'm working on fixing it

**Comment by metaprime** [comment]
@cyian-1756 Ah, okay. That makes more sense. Easy to blame the first potential problem you see I guess. Thanks for looking into it!

**Comment by cyian-1756** [comment]
@metaprime my latest commit should have everything working again

**Pull request merged**